applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/ipsets.js

```diff
@@ -0,0 +1,217 @@
+'use strict';
+'require view';
+'require uci';
+'require form';
+'require firewall';
+'require tools.firewall as fwtool';
+
+
+return view.extend({
+
+	load: function() {
+		return Promise.all([
+			uci.load('firewall')
+		]);
+	},
+
+	render: function(data) {
+		if (fwtool.checkLegacySNAT())
+			return fwtool.renderMigration();
+		else
+			return this.renderIPsets(data);
+	},
+
+	renderIPsets: function(data) {
+		var boardinfo = data[0],
+		    m, s, o;
+
+		m = new form.Map('firewall', _('Firewall - IP sets'),
+			_('firewall4 supports referencing and creating IP sets to simplify matching of large address lists without the need to create one rule per item to match. Port ranges in ipsets are unsupported by firewall4.<br />' + 
+				'<a href="https://ipset.netfilter.org/ipset.man.html">Reference</a>.'));
+
+
+		if (L.hasSystemFeature('firewall4')) {
```

Comment:
This does not belong here imho, if at all it should appear on the zone overview page.